psl-core/src/main/java/org/linqs/psl/database/Database.java

```diff
@@ -293,6 +288,56 @@ public Partition getWritePartition() {
         return writePartition;
     }
 
+    /**
+     * Output all random variables to stdout in a human readable format: Foo('a', 'b') = 1.0.
+     */
+    public void outputRandomVariableAtoms() {
+        for (StandardPredicate openPredicate : parentDataStore.getRegisteredPredicates()) {
+            for (GroundAtom atom : getAllGroundRandomVariableAtoms(openPredicate)) {
+                System.out.println(atom.toString() + " = " + atom.getValue());
+            }
+        }
+    }
+
+    /**
+     * Output all random variables in a tab separated format.
+     */
+    public void outputRandomVariableAtoms(String outputDirectoryPath) {
+        File outputDirectory = new File(outputDirectoryPath);
+
+        // mkdir -p
+        outputDirectory.mkdirs();
+
+        for (StandardPredicate predicate : parentDataStore.getRegisteredPredicates()) {
+            if (getAllGroundRandomVariableAtoms(predicate).size() == 0) {
```

Comment:
Since `getAllGroundRandomVariableAtoms` is not free, we gotta be a little careful here.
I am thinking something like:

```
if (isClosed(predciate)) {
    continue;
}

List<RandomVariableAtom> atoms = getAllGroundRandomVariableAtoms(predicate);
if (atoms.size() == 0) {
    continue;
}

...

for (GroundAtom atom : atoms) {

...

```